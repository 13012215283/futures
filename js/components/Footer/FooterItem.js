import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet, Text } from 'react-native';

import { colors, sizes } from '../themes';
import { px2dp } from '../../utils/px2dp';
import noop from '../../utils/noop';

export default function FooterItem({ text, onPress, type }) {
  return (
    <View>
      <Text
        style={type === 'separating' ? styles.separating : styles.footerText}
        onPress={onPress}
      >
        {text}
      </Text>
    </View>
  );
}

FooterItem.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
  onPress: PropTypes.func,
};

FooterItem.defaultProps = {
  type: '',
  onPress: noop,
};

const styles = StyleSheet.create({
  footerText: {
    color: colors[1004],
    fontSize: sizes.f1,
  },
  separating: {
    margin: px2dp(32),
  },
});
