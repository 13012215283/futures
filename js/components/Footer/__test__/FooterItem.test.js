import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FooterItem from '../FooterItem';

describe('</FooterItem/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FooterItem text="忘记密码" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as separator', () => {
    const tree = renderer
      .create(<FooterItem text="|" type="separating" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('simulates events correctly ', () => {
    let click = false;
    const tree = shallow(
      <FooterItem
        text="click！"
        onPress={() => {
          click = true;
        }}
      />
    );

    tree.find('Text').simulate('press');
    expect(click).toBe(true);
  });
});

it('renders correctly as separator', () => {
  const tree = renderer
    .create(<FooterItem text="|" type="separating" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
