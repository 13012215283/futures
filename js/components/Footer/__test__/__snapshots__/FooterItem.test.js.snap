// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</FooterItem/> renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#0040ff",
        "fontSize": 24,
      }
    }
  >
    忘记密码
  </Text>
</View>
`;

exports[`</FooterItem/> renders correctly as separator 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "margin": 32,
      }
    }
  >
    |
  </Text>
</View>
`;

exports[`renders correctly as separator 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "margin": 32,
      }
    }
  >
    |
  </Text>
</View>
`;
